src/World.cc

```diff
@@ -283,21 +285,35 @@ Errors World::Load(sdf::ElementPtr _sdf, const ParserConfig &_config)
     std::string frameName = frame.Name();
     if (frameNames.count(frameName) > 0)
     {
-      frameName += "_frame";
-      int i = 0;
-      while (frameNames.count(frameName) > 0)
+      // This frame has a name collision
+      if (sdfVersion < gz::math::SemanticVersion(1, 7))
       {
-        frameName = frame.Name() + "_frame" + std::to_string(i++);
+        // This came from an old file, so try to workaround by renaming frame
+        frameName += "_frame";
+        int i = 0;
+        while (frameNames.count(frameName) > 0)
+        {
+          frameName = frame.Name() + "_frame" + std::to_string(i++);
+        }
+        std::stringstream ss;
+        ss << "Frame with name [" << frame.Name() << "] "
+           << "in world with name [" << this->Name() << "] "
+           << "has a name collision, changing frame name to ["
+           << frameName << "].";
```

Comment:
related to https://github.com/gazebosim/sdformat/issues/1312

I can revert this if desired